interim-jan-2021/resolver-behavior-self-description.md

```diff
@@ -0,0 +1,27 @@
+# Resolver behavior self-description
+
+## Defining local-only namespaces
+* Express namespaces which only this resolver can resolve
+* Ex 1: Enterprise resolvers serving corporation-specific namespaces
+* Ex 2: Public Wi-Fi or cellular networks offering network-local services
+
+## Defining per-namespace optimization
+* Express namespaces for which this resolver provides preferable resolutions
+* Ex 1: ISP routes public content requests to network caches
+* Ex 2: Cellular provider charges less for content from specific servers
+* Ex 3: Public resolver designated to serve a namespace to limit parties privy to resolution
+
+## Defining answer modification behavior
+* Express how query answers are modified by the resolver (but not necessarily _why_)
+* Ex 1: Will not resolve names breaching resolver policy 
```

Comment:
Surely no resolver will resolve names that breach its policy?  This seems tautological, so I don't think we need a way to express it.

Reply:
My intended point is to allow resolvers to communicate whether they do any modification of answers at all. Some resolvers don't such as those qualifying for Mozilla's TRR program. Some block ad domains. Some don't. 

I think this is a scenario that has gathered sufficient interest in the WG to document as a requirement for drafts to meet. I think it's sufficiently different from policy (describes resolver behavior without mandating client behavior) to not be a non-goal. 

Open to discussion.

Reply:
> My intended point is to allow resolvers to communicate whether they do any modification of answers at all.

I find this difficult to reason about because (a) it's a forward-looking statement and (b) even resolvers that don't normally reject any queries probably will do so under some exceptional circumstance (e.g. a DoS attack).

I think the EDE "which codes to expect" requirement is a better way to draw this distinction.

Reply:
It's fair to say that the EDE line could cover this use case with a much cleaner architecture. I'll remove this line.